code/modules/spacepods/pod_fab.dm

```diff
@@ -0,0 +1,489 @@
+	var/screen = "main"
+	var/temp
+	var/list/part_sets = list(
+								"Pod Weaponry",
+								"Pod Armor",
+								"Pod Cargo",
+								"Pod Parts",
+								"Pod Frame",
+								"Misc",
+								)
+
+/obj/machinery/spod_part_fabricator/Initialize()
+	files = new /datum/research(src) //Setup the research data holder.
+	materials = new(src, list(MAT_METAL, MAT_GLASS, MAT_SILVER, MAT_GOLD, MAT_DIAMOND, MAT_PLASMA, MAT_URANIUM, MAT_BANANIUM, MAT_TITANIUM, MAT_BLUESPACE))
+	return ..()
+
+/obj/machinery/spod_part_fabricator/RefreshParts()
+	var/T = 0
+
+	//maximum stocking amount (default 300000, 600000 at T4)
+	for(var/obj/item/weapon/stock_parts/matter_bin/M in component_parts)
+		T += M.rating
+	materials.max_amount = (200000 + (T*50000))
+
+	//resources adjustment coefficient (1 -> 0.85 -> 0.7 -> 0.55)
+	T = 1.15
+	for(var/obj/item/weapon/stock_parts/micro_laser/Ma in component_parts)
+		T -= Ma.rating*0.15
+	component_coeff = T
+
+	//building time adjustment coefficient (1 -> 0.8 -> 0.6)
+	T = -1
+	for(var/obj/item/weapon/stock_parts/manipulator/Ml in component_parts)
+		T += Ml.rating
+	time_coeff = round(initial(time_coeff) - (initial(time_coeff)*(T))/5,0.01)
+
+
+/obj/machinery/spod_part_fabricator/check_access(obj/item/weapon/card/id/I)
+	if(istype(I, /obj/item/device/pda))
+		var/obj/item/device/pda/pda = I
+		I = pda.id
+	if(!istype(I) || !I.access) //not ID or no access
+		return 0
+	for(var/req in req_access)
+		if(!(req in I.access)) //doesn't have this access
+			return 0
+	return 1
+
+/obj/machinery/spod_part_fabricator/emag_act()
+	if(emagged)
+		return
+	emagged = TRUE
+	req_access = null
+	say("DB error \[Code 0x00F1\]")
+	sleep(10)
+	say("Attempting auto-repair...")
+	sleep(15)
+	say("User DB corrupted \[Code 0x00FA\]. Truncating data structure...")
+	sleep(30)
+	say("User DB truncated. Please contact your Nanotrasen system operator for future assistance.")
+
+
+/obj/machinery/spod_part_fabricator/proc/output_parts_list(set_name)
+	var/output = ""
+	for(var/v in files.known_designs)
+		var/datum/design/D = files.known_designs[v]
+		if(D.build_type & PODFAB)
+			if(!(set_name in D.category))
+				continue
+			output += "<div class='part'>[output_part_info(D)]<br>\["
+			if(check_resources(D))
+				output += "<a href='?src=\ref[src];part=[D.id]'>Build</a> | "
+			output += "<a href='?src=\ref[src];add_to_queue=[D.id]'>Add to queue</a>\]\[<a href='?src=\ref[src];part_desc=[D.id]'>?</a>\]</div>"
+	return output
+
+/obj/machinery/spod_part_fabricator/proc/output_part_info(datum/design/D)
+	var/output = "[initial(D.name)] (Cost: [output_part_cost(D)]) [get_construction_time_w_coeff(D)/10]sec"
+	return output
+
+/obj/machinery/spod_part_fabricator/proc/output_part_cost(datum/design/D)
+	var/i = 0
+	var/output
+	for(var/c in D.materials)
+		output += "[i?" | ":null][get_resource_cost_w_coeff(D, c)] [material2name(c)]"
+		i++
+	return output
+
+/obj/machinery/spod_part_fabricator/proc/output_available_resources()
+	var/output
```

Comment:
```var/list/output = list()```

Going to explain this one now it's occurred so many times.
Whenever you see code that ```+=```'s to a string, especially in a loop, should be done as a ```list()```, and then do ```.Join()``` on that list at the end to return the final string